src/averageAges.js

```diff
@@ -55,7 +81,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child =>
+      people.find(person => person.name === child.mother)
+      && child.sex === 'm')
+    : people.filter(child =>
+      people.find(person => person.name === child.mother));
```

Comment:
why repeating same logic?
```suggestion
  const children = people.filter(child =>
      people.find(person => person.name === child.mother)
      && (onlyWithSon ? child.sex === 'm') : true);
```